x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -68,9 +68,16 @@ def pipeline_configs
             return @cached_pipelines
           end
         end
+
         es_version = get_es_version
         fetcher = get_pipeline_fetcher(es_version)
-        fetcher.fetch_config(es_version, pipeline_ids, client)
+
+        begin
+          fetcher.fetch_config(es_version, pipeline_ids, client)
+        rescue LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError => e
+          return [] if e.response_code == 404
```

Comment:
Will this early return cause a difference in behavior whereby the `@cached_pipelines` are not updated on 404 now? Does that matter?

Reply:
It doesn't matter. The pipeline is empty anyway and the cache is purely optional

Reply:
Reviewing the flow, I decide to remove the cache because the existing pipelines can keep running without cache when error is raised. It was there to prevent cleaning up the running pipelines.